src/eu/deltasource/internship/ecosystem/ecosystems/Ecosystem.java

```diff
@@ -0,0 +1,184 @@
+                    } else {
+                        carnivores.get(i).eat(herbivores.get(chosenHerbivore), 1);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks how many animals are in the group, split the prey equally to them and after that they eat it.
+     * @param i takes the carnivore index.
+     * @param chosenHerbivore takes the herbivore that is attacked.
+     */
+    private static void groupCarnivoreAttack(int i, int chosenHerbivore) {
+        int numberOfAnimalsInTHeGroup;
+        numberOfAnimalsInTHeGroup = 0;
+        for (int j = 0; j < carnivores.size(); j++) {
+            if (carnivores.get(i).getClass() == carnivores.get(j).getClass()) {
+                numberOfAnimalsInTHeGroup++;
+            }
+            for (int b = 0; b < carnivores.size(); b++) {
+                if (carnivores.get(i).getClass() == carnivores.get(b).getClass()) {
+                    carnivores.get(b).eat(herbivores.get(chosenHerbivore), numberOfAnimalsInTHeGroup);
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks if the animals reach the maximum level of starvation and if it does set his alive variable to false.
+     */
+    private static void starvationCheck() {
+        for (int i = 0; i < carnivores.size(); i++) {
+            if (carnivores.get(i).getHungerLevel() >= 100) {
+                carnivores.get(i).setAlive(false);
+            }
+        }
+    }
+
+    /**
+     * Checks if the animals reached the maximum age and if its true it set his alive variable to false;
+     */
+    private static void ageCheck() {
+        for (int i = 0; i < carnivores.size(); i++) {
+            if (carnivores.get(i).getAge() == carnivores.get(i).getMaxAge()) {
+                carnivores.get(i).setAlive(false);
+            }
+        }
+
+        for (int i = 0; i < herbivores.size(); i++) {
+            if (herbivores.get(i).getAge() == herbivores.get(i).getMaxAge()) {
+                herbivores.get(i).setAlive(false);
+            }
+        }
+    }
+
+    /**
+     * Checks if the animals are ready to reproduce and if they are it calls the reproduce method them.
+     */
+    private static void reproduce() {
+        for (int i = 0; i < carnivores.size(); i++) {
+            if (carnivores.get(i).getAge() % carnivores.get(i).getReproductionRate() == 0) {
+                carnivores.add((Carnivore) carnivores.get(i).reproduce());
+            }
+        }
+
+        for (int i = 0; i < herbivores.size(); i++) {
+            if (herbivores.get(i).getAge()  % herbivores.get(i).getReproductionRate() == 0) {
+                herbivores.add((Herbivore) herbivores.get(i).reproduce());
+            }
+        }
+    }
+
+    /**
+     * Goes through all animals and increase their age by 1.
+     */
+    private static void age() {
+        for (int i = 0; i < carnivores.size(); i++) {
+            carnivores.get(i).setAge(carnivores.get(i).getAge() + 1);
```

Comment:
I would rather provide a method that lets the animal grow older, instead of setting the age.